File: Mediator/PHP/Controller.php
<?php
  class Controller implements Mediator {
    protected $model;

    protected $view;

    public function setColleague($model, $view) {
      $this->model = $model;
      $this->view = $view;
    }

    public function addUser($user) {
      $this->model->addUser($user);
    }

    public function getUserList() {
      return $this->model->getUserList();
    }

    public function deleteUser($user) {
      $this->model->deleteUser($user);
    }

    public function showResult($result) {
      $this->view->showResult($result);
    }
  }

File: Mediator/Go/mediator/Controller.go
package mediator

type Controller struct {
  model *Model
  view *View
}

func (this *Controller) SetColleague(model *Model, view *View) {
  this.model = model
  this.view = view
}

func (this *Controller) AddUser(user string) {
  this.model.AddUser(user)
}

func (this *Controller) GetUserList() []string {
  return this.model.GetUserList()
}

func (this *Controller) DeleteUser(user string) {
  this.model.DeleteUser(user)
}

func (this *Controller) ShowResult(result string) {
  this.view.ShowResult(result)
}
